bindings/python/pymongocrypt/mongocrypt.py

```diff
@@ -753,3 +792,23 @@ def __init__(self, ctx, filter, provider, master_key, callback):
             # Destroy the context on error.
             self._close()
             raise
+
+
+def _ask_for_kms_credentials(kms_providers):
+    """Get on-demand kms credentials.
+
+    This is a separate function so it can be overridden in unit tests."""
+    if 'aws' not in kms_providers:
+        return
+    if len(kms_providers['aws']):
```

Comment:
If we already provided non-empty aws creds to libmongocrypt is this state even possible to reach?

Reply:
This was in preparation for other types of on-demand credentials.